addon/-private/formatters/format-number.ts

```diff
@@ -27,33 +27,29 @@ const NUMBER_OPTIONS = [
   'signDisplay',
   'unitDisplay',
   'unit',
-];
+] as readonly (keyof Intl.NumberFormatOptions)[];
 
 /**
  * @private
  * @hide
  */
-export default class FormatNumber extends Formatter {
-  static type = 'number';
+export default class FormatNumber extends Formatter<Intl.NumberFormatOptions> {
+  static readonly type = 'number';
+
+  createNativeFormatter = memoize((locales: string | string[], options: Intl.NumberFormatOptions) => {
+    return new Intl.NumberFormat(locales, options);
+  });
 
   get options() {
     return NUMBER_OPTIONS;
   }
 
-  constructor(config) {
-    super(config);
-
-    this.createNativeFormatter = memoize((locales, options) => {
-      return new Intl.NumberFormat(locales, options);
-    });
-  }
-
-  format(locale, value, formatOptions) {
+  format(locale: string | string[], value: number, formatOptions: Intl.NumberFormatOptions & BaseOptions): string {
     const formatterOptions = this.readOptions(formatOptions);
 
     this.validateFormatterOptions(locale, formatterOptions);
     const formatterInstance = this.createNativeFormatter(locale, formatterOptions);
 
-    return formatterInstance.format(value, formatOptions);
+    return formatterInstance.format(value);
```

Comment:
According to MDN, there is no second parameter:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/DateTimeFormat/format#Syntax